crates/turborepo-filewatch/src/globwatcher.rs

```diff
@@ -24,26 +25,53 @@ pub struct GlobSet {
     exclude: Any<'static>,
 }
 
+#[derive(Debug, Error)]
+pub struct GlobError {
+    // Boxed to minimize error size
+    underlying: Box<wax::BuildError>,
+    raw: String,
+}
+
+impl Display for GlobError {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "{}: {}", self.underlying, self.raw)
+    }
+}
+
+fn compile_glob(raw: &str) -> Result<Glob<'static>, GlobError> {
+    Glob::from_str(raw)
+        .map(|g| g.to_owned())
+        .map_err(|e| GlobError {
+            underlying: Box::new(e),
+            raw: raw.to_owned(),
+        })
+}
+
 impl GlobSet {
     pub fn from_raw(
         raw_includes: Vec<String>,
         raw_excludes: Vec<String>,
-    ) -> Result<Self, wax::BuildError> {
+    ) -> Result<Self, GlobError> {
         let include = raw_includes
             .into_iter()
             .map(|raw_glob| {
-                let glob = Glob::from_str(&raw_glob)?.to_owned();
+                let glob = compile_glob(&raw_glob)?;
                 Ok((raw_glob, glob))
             })
-            .collect::<Result<HashMap<_, _>, wax::BuildError>>()?;
+            .collect::<Result<HashMap<_, _>, GlobError>>()?;
         let excludes = raw_excludes
-            .into_iter()
+            .iter()
             .map(|raw_glob| {
-                let glob = Glob::from_str(&raw_glob)?.to_owned();
+                let glob = compile_glob(raw_glob)?;
                 Ok(glob)
             })
-            .collect::<Result<Vec<_>, wax::BuildError>>()?;
-        let exclude = wax::any(excludes)?.to_owned();
+            .collect::<Result<Vec<_>, GlobError>>()?;
+        let exclude = wax::any(excludes)
```

Comment:
Reading the doc comment of `wax::any`, since we're passing `Glob`s I think we'll only fail if the combined excluded globs end up being [10 MB](https://docs.rs/regex/latest/src/regex/builders.rs.html#53). If we end up hitting that, we probably shouldn't print out that much data to the terminal and just say that the combined excludes are too large.

Reply:
I went to switch this to something like "globs elided, combined excludes are too large", but decided to back out the change. I think in the (hopefully extremely rare) case when this happens, we still want to know what those globs are, as a pointer to where they came from, vs not having any idea.

If in the future we have some provenance information, like "the globs from line 5 of `turbo.json` are too big", I think it would be better to just do that, but while we're lacking that information, I think explicit is better here so that a user / we have a chance at tracking down where the globs came from.